src/dot/mod.rs

```diff
@@ -0,0 +1,99 @@
+use verkle_trie::proof::UpdateHint;
+use ark_serialize::{CanonicalSerialize};
+// use verkle_trie::EdwardsProjective;
+use std::borrow::Cow;
+
+use std::io::Write;
+
+type Nd = usize;
+type Ed<'a> = &'a (usize, usize, u8);
+struct Graph { nodes: Vec<String>, edges: Vec<(usize,usize,u8)> }
+
+
+pub fn render_to<W: Write>(output: &mut W, data: &UpdateHint) {
+    let mut nodes = vec![];
+    let mut previous_items = Vec::<(Vec::<u8>, String)>::new();
+    let mut edges = vec![];
+    let mut my_index = 0;
+
+    for (path, comm) in data.commitments_by_path.clone() {
```

Comment:
do you really need to clone here? can't you do `iter()`? and clone only the specific values if needed?